pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/HashJoinOperator.java

```diff
@@ -82,6 +90,8 @@ public HashJoinOperator(Operator<TransferableBlock> leftTableOperator, Operator<
     _isHashTableBuilt = false;
     _broadcastHashTable = new HashMap<>();
     _upstreamErrorBlock = null;
+    // TODO: Consider figuring out a better way to support join data left empty case.
+    _hasSentData = false;
```

Comment:
i dont think this is actually a bug on the left empty or right empty. processing of a left or right empty block is the same as not-processing any block. and end-of-stream is a pass through. 

I agree we should consolidate the behavior but let's not address this issue here and split into a separate PR